### PR TITLE
docs: Improve examples

### PR DESCRIPTION
**Pull request by baolsen**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on: [workflow_dispatch, pull_request]
 env:
   AWS_REGION: af-south-1
   AWS_ROLE: cat-genrl-prd-infra-gh-tf-aws-github-runners
-  TERRAFORM_MIN_SUPPORTED_VERSION: 0.13.1
+  TERRAFORM_MIN_SUPPORTED_VERSION: 0.15.0
```
Comment (baolsen) [inline .github/workflows/main.yml:9]: The lambda module we use doesn't support TF versions before 0.15.0
So our module was actually incorrectly saying we can support that version. Fixed now.

File: examples/email/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<!-- BEGIN_TF_DOCS -->
+----
```
Comment (baolsen) [inline examples/email/README.md:2]: Examples now automatically document themselves :D

File: examples/email/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+## main.tf
+```hcl
+module "example" {
+
+  # Update source and version as needed.
+  # TODO why cant conftest do this?
```
Comment (baolsen) [inline examples/email/README.md:8]: I'll address this TODO later

File: iam.tf [hunk 4]
```diff
@@ -3,21 +3,22 @@ resource "aws_iam_policy" "cost_explorer_access_policy" {
...
+      [for aws_sns_topic in aws_sns_topic.cost_notifier :
+        {
+          Effect = "Allow"
+          Action = [
+            "sns:Publish"
+          ]
+          Resource = aws_sns_topic.arn
+      }]
+    )
   })
```
Comment (baolsen) [inline iam.tf:23]: Fix was needed because `aws_sns_topic.cost_notifier` is created with a count

File: main.tf [hunk 5]
```diff
@@ -38,28 +38,40 @@ locals {
 
 #tfsec:ignore:aws-lambda-enable-tracing
```
Comment (baolsen) [inline main.tf:39]: Tidied up this file a bit

**Review by nomfulavelisa — APPROVED** [review]

**Pull request merged**